src/tests/ffi.cpp

```diff
@@ -2510,8 +2512,14 @@ TEST_F(rnp_tests, test_ffi_add_userid)
     assert_int_equal(
       RNP_SUCCESS, rnp_key_add_uid(key_handle, new_userid, "SHA256", 2147317200, 0x00, false));
 
+    // add with default and non-default hash algorithms
+    assert_rnp_success(
+      rnp_key_add_uid(key_handle, default_hash_userid, NULL, 2147317200, 0, false));
+    assert_rnp_success(
+      rnp_key_add_uid(key_handle, ripemd_hash_userid, "RIPEMD160", 2147317200, 0, false));
+
     assert_rnp_success(rnp_key_get_uid_count(key_handle, &count));
-    assert_int_equal(2, count);
+    assert_int_equal(4, count);
```

Comment:
Please also check that default algorithm was used in first case. This can be done via `rnp_key_get_uid_handle_at()`, `rnp_uid_get_signature_at()` and `rnp_signature_get_hash_alg()`.